openselery/openselery.py

```diff
@@ -30,27 +35,40 @@ def __init__(self, config, silent=False):
         self.initialize()
 
     def __del__(self):
-        self.logNotify(
-            "Feel free to visit us @ https://github.com/protontypes/openselery")
-        if hasattr(self, 'receiptStr') and self.receiptStr != "":
+        self.logNotify("Feel free to visit us @ https://github.com/protontypes/openselery")
+        print("=======================================================")
+
+    def finish(self, receiptFilePath):
+        success = True
+        if receiptFilePath:
             self.logNotify("Done")
         else:
-            self.logNotify("receipt missing")
-            self.logNotify("Failed!")
+            self.logWarning("Receipt missing")
+            self.logWarning("Failed!")
+            success = False
+        return success
 
     def initialize(self):
         self.logNotify("Initializing OpenSelery")
 
-        # return openselery version
-        selerypath = os.path.realpath(__file__)
-        # return openselery version
-        self.log("OpenSelery HEAD sha [%s]" % git_utils.get_head_sha(selerypath))
-        self.log("OpenSelery last tag [%s]" % git_utils.get_lastest_tag(selerypath))
+        self.seleryPackageInfo = os_utils.getPackageInfo("openselery")
+        self.log("OpenSelery version [%s]" % self.seleryPackageInfo["version"])
 
         self.log("Preparing Configuration")
-        # apply yaml config to our configuration if possible
-        self.log("Loading configuration [%s]" % self.config.config_path)
-        self.loadYaml(self.config.config_path)
+        ### find all configs in potentially given configdir
+        foundConfigs = []
```

Comment:
I am just wondering, is there a use-case for merging multiple configs? It can be confusing with overwrites and the order of configs.

Reply:
@cornerman dunno, I added it because configuration management and separation of concerns is something taht comes up as soon as something get's "bigger". 

In the case of configurations, dividing configs up into multiple different configs for different matters is something anyone could do (for example having 3 separate directories somewhere containing 3 different sets of rule sets). 

because everything is dict based, i do agree with the "config order" problem, but that is not really an issue because your configuration set is either valid (and whatever you want it to be) or it is not. The same goes for nearly every linux tool out there.